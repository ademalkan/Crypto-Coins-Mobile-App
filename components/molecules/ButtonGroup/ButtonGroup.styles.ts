import { StyleSheet } from "react-native";

export default StyleSheet.create({
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginBottom: 16,
  },
})