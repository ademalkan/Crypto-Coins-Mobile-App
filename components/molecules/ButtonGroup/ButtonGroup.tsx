import React, { useMemo } from 'react';
import { View } from 'react-native';
import RangeButton from '../../atoms/RangeButton/RangeButton';
import styles from './ButtonGroup.styles';

interface ButtonGroupProps {
  buttons: { label: string; onPress: () => void; selected?: boolean }[];
}

const ButtonGroup: React.FC<ButtonGroupProps> = ({ buttons }) => {
  const memoizedButtons = useMemo(() => buttons, [buttons]);

  return (
    <View style={styles.buttonContainer}>
      {memoizedButtons.map((button, index) => (
        <RangeButton key={index} label={button.label} onPress={button.onPress} selected={button.selected} />
      ))}
    </View>
  );
};

export default ButtonGroup;
