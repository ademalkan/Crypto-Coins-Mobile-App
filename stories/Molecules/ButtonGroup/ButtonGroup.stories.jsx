import React from "react";
import { View } from "react-native";
import ButtonGroup from "../../../components/molecules/ButtonGroup/ButtonGroup";

export default {
  title: "Molecules/ButtonGroup",
  component: ButtonGroup,
};

const buttons = [
  { label: "Button 1", onPress: () => console.log("Button 1 pressed") },
  {
    label: "Button 2",
    onPress: () => console.log("Button 2 pressed"),
    selected: true,
  },
  { label: "Button 3", onPress: () => console.log("Button 3 pressed") },
];

export const Default = () => (
  <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
    <ButtonGroup buttons={buttons} />
  </View>
);

export const CustomStyle = () => (
  <View
    style={{
      flex: 1,
      justifyContent: "center",
      alignItems: "center",
      backgroundColor: "#f0f0f0",
    }}
  >
    <ButtonGroup buttons={buttons} />
  </View>
);

export const DisabledButton = () => (
  <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
    <ButtonGroup
      buttons={[
        {
          label: "Button 1",
          onPress: () => console.log("Button 1 pressed"),
          selected: true,
        },
        {
          label: "Button 2",
          onPress: () => console.log("Button 2 pressed"),
          disabled: true,
        },
        { label: "Button 3", onPress: () => console.log("Button 3 pressed") },
      ]}
    />
  </View>
);
