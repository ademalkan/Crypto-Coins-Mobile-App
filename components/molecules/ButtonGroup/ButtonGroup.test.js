import React from "react";
import renderer from "react-test-renderer";
import ButtonGroup from "./ButtonGroup";

describe("ButtonGroup", () => {
  it("should render the buttons correctly", () => {
    const buttons = [
      { label: "Button 1", onPress: jest.fn(), selected: true },
      { label: "Button 2", onPress: jest.fn(), selected: false },
      { label: "Button 3", onPress: jest.fn(), selected: false },
    ];

    const component = renderer.create(<ButtonGroup buttons={buttons} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
