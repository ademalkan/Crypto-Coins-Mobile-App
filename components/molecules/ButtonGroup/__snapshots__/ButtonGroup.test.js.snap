// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render the buttons correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ccc",
        "borderColor": "#ccc",
        "borderRadius": 4,
        "borderWidth": 1,
        "marginHorizontal": 8,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      Button 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#ccc",
        "borderRadius": 4,
        "borderWidth": 1,
        "marginHorizontal": 8,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      Button 2
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#ccc",
        "borderRadius": 4,
        "borderWidth": 1,
        "marginHorizontal": 8,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      Button 3
    </Text>
  </View>
</View>
`;
